src/api/voting.ts

```diff
@@ -206,41 +229,79 @@ export class VotingApi {
      * @param processId
      * @param gateway
+     * @param checkArchive
      */
-    static getProcessSummary(processId: string, gateway: IGateway | IGatewayPool): Promise<IProcessSummary> {
+    static getProcessSummary(processId: string, gateway: IGateway | IGatewayPool, checkArchive: boolean = false): Promise<IProcessSummary> {
         if (!processId) return Promise.reject(new Error("Empty process ID"))
         else if (!gateway) return Promise.reject(new Error("Invalid Gateway object"))
 
         return gateway.sendRequest({ method: "getProcessSummary", processId })
+            .catch((error) => {
+                const message = error.message ? "Could not retrieve the process info: " + error.message : "Could not retrieve the process info"
+                if (checkArchive && message.includes("No data found for this key")) {
+                    return this.getArchiveProcessSummary(processId, gateway, message)
+                }
+                throw new Error(message)
+            })
             .then((response) => {
-                if (!response.ok) throw new Error(response.message || null)
                 const { processSummary } = response
 
                 return {
                     entityId: "0x" + processSummary.entityId,
                     envelopeHeight: processSummary.envelopeHeight,
-                    status: VochainProcessStatus[processSummary.state as string],
+                    status: processSummary.status ? processSummary.status : VochainProcessStatus[processSummary.state as string],
                     envelopeType: processSummary.envelopeType || {},
                     startBlock: processSummary.startBlock,
-                    endBlock: processSummary.startBlock + processSummary.blockCount,
+                    endBlock: processSummary.endBlock ? processSummary.endBlock : processSummary.startBlock + processSummary.blockCount,
                     entityIndex: processSummary.entityIndex,
                     metadata: processSummary.metadata,
                     sourceNetworkId: processSummary.sourceNetworkID || processSummary.sourceNetworkId,
                 } as IProcessSummary
             })
-            .catch((error) => {
-                const message = error.message ? "Could not retrieve the process info: " + error.message : "Could not retrieve the process info"
-                throw new Error(message)
+    }
+
+    /**
+     * Fetch the headers of the given processId on the archive.
+     *
+     * @param processId
+     * @param gateway
+     * @param errorMessage
+     */
+    private static getArchiveProcessSummary(processId: string, gateway: IGateway | IGatewayPool, errorMessage: string): Promise<DVoteGatewayResponseBody> {
+        return GatewayArchive.getProcessFromArchive(processId, gateway, errorMessage)
+            .then((response: DVoteGatewayResponseBody) => {
+                // TODO add only `envelopeHeight`
+                return {
```

Comment:
Why can't this function return its own schema right away... and let others using it do the adaptation? Why should this function return something to accommodate for schemas that only our consumers/callers will know for sure?

Reply:
Shouldn't this function be part of `GatewayArchive` ? It seems to add little here